### PR TITLE
[FIX] Remove rows with empty abstract before running LDAModel

### LINKED ISSUE #366: LDAModel does not drop studies without texts
If MALLET is setup (`model = annotate.lda.LDAModel(dset_abstract.texts, text_column='abstract', n_topics=200)`) with a `dset_abstract.texts` with at least one study without the abstract, the function will give the error:

```
Traceback (most recent call last):
  File "generate_cog-atlas.py", line 101, in <module>
    model = annotate.lda.LDAModel(dset_abstract.texts, text_column='abstract', n_topics=200)
  File "/opt/anaconda3/lib/python3.8/site-packages/nimare/annotate/lda.py", line 103, in __init__
    fo.write(text)
TypeError: write() argument must be str, not float
```

This currently happens for study [ 9728909 ](https://pubmed.ncbi.nlm.nih.gov/9728909/) and [20685394](https://pubmed.ncbi.nlm.nih.gov/20677377/) from Neurosynth (abstract for 9728909 is not available, and 20685394 was deleted because is a duplicated of 20677377).

### PR DESCRIPTION
**Pull request by JulioAPeraza**
Closes #366 .

Changes proposed in this pull request:

- Add code, in the initialization of LDAModel, to automatically remove rows with empty text_column (abstract).

### CODE REVIEW

File: nimare/annotate/lda.py [hunk 1]
```diff
@@ -92,6 +92,14 @@ def __init__(
         if not op.isdir(text_dir):
             LGR.info("Texts folder not found. Creating text files...")
             os.mkdir(text_dir)
+            # Remove rows with empty text cells
+            #text_df.dropna(subset=[text_column],inplace=True)
```
Comment (tsalo) [inline nimare/annotate/lda.py:96]: Does this not work?

File: nimare/annotate/lda.py [hunk 2]
```diff
@@ -92,6 +92,14 @@ def __init__(
...
+            orig_ids = text_df["id"].tolist()
+            text_df = text_df.fillna("")
+            keep_ids = text_df.loc[text_df[text_column] != "", "id"]
+            text_df = text_df.loc[text_df["id"].isin(keep_ids)]
+            if len(keep_ids) != len(orig_ids):
+                LGR.info("Retaining {0}/{1} studies".format(len(keep_ids), len(orig_ids)))
             for id_ in text_df["id"].values:
```
Comment (tsalo) [inline nimare/annotate/lda.py:103]: Could you add a new line here?

File: nimare/annotate/lda.py (hunk 1 above)
Comment (JulioAPeraza) [reply nimare/annotate/lda.py:96]: Yes, it works:
```
orig_ids = text_df["id"].tolist()
text_df.dropna(subset=[text_column],inplace=True)
keep_ids = text_df["id"].tolist()
```

Comment (tsalo) [reply nimare/annotate/lda.py:96]: I'd go with that then, although I'd prefer to operate on a copy instead (i.e., drop `inplace=True` and assign to a fresh `text_df`).

**Review by tsalo — APPROVED** [review]

**Comment by tsalo** [comment]
@JulioAPeraza Is there anything else you want to do in this PR, or should I merge?

**Comment by JulioAPeraza** [comment]
You can merge it. Thanks!

**Pull request merged**